api/service/explorer/structs.go

```diff
@@ -79,3 +82,51 @@ func GetTransaction(tx *types.Transaction, addressBlock *types.Block) *Transacti
 		Type:      "",
 	}
 }
+
+// StakingTransaction ...
+type StakingTransaction struct {
+	ID        string   `json:"id"`
+	Timestamp string   `json:"timestamp"`
+	From      string   `json:"from"`
+	To        string   `json:"to"`
+	Value     *big.Int `json:"value"`
+	Bytes     string   `json:"bytes"`
+	Data      string   `json:"data"`
+	GasFee    *big.Int `json:"gasFee"`
+	FromShard uint32   `json:"fromShard"`
+	ToShard   uint32   `json:"toShard"`
+	Type      string   `json:"type"`
+}
+
+// GetStakingTransaction ...
+func GetStakingTransaction(tx *staking.StakingTransaction, addressBlock *types.Block) *StakingTransaction {
+	msg, err := core2.StakingToMessage(tx, addressBlock.Header().Number())
+	if err != nil {
+		utils.Logger().Error().Err(err).Msg("Error when parsing tx into message")
+	}
+	gasFee := big.NewInt(0)
+	gasFee = gasFee.Mul(tx.GasPrice(), new(big.Int).SetUint64(tx.Gas()))
+	to := ""
+	if msg.To() != nil {
+		if to, err = common.AddressToBech32(*msg.To()); err != nil {
+			return nil
+		}
+	}
+	from := ""
+	if from, err = common.AddressToBech32(msg.From()); err != nil {
+		return nil
+	}
+	return &StakingTransaction{
+		ID:        tx.Hash().Hex(),
+		Timestamp: strconv.Itoa(int(addressBlock.Time().Int64() * 1000)),
+		From:      from,
+		To:        to,
+		Value:     msg.Value(),
+		Bytes:     strconv.Itoa(int(tx.Size())),
+		Data:      hex.EncodeToString(tx.Data()),
+		GasFee:    gasFee,
+		FromShard: tx.ShardID(),
+		ToShard:   0,
+		Type:      "",
```

Comment:
Should you assign the msg.Type() here instead of "", as it is already computed?

Reply:
This type actually refers to whether it is a sent or received txn. It was how it was done for regular txns and just duplicated here. But since staking txns are all just sent txns, I might as well update the type enum to include staking txn types and use this same field.

Will make the change accordingly.